src/runtime/builtin_modules/builtins.cpp

```diff
@@ -145,30 +145,68 @@ extern "C" Box* any(Box* container) {
     return boxBool(false);
 }
 
-extern "C" Box* min(Box* arg0, BoxedTuple* args) {
+Box* min_max(Box* arg0, BoxedTuple* args, BoxedDict* kwargs, int opid) {
     assert(args->cls == tuple_cls);
+    assert(kwargs->cls == dict_cls);
 
-    Box* minElement;
+    bool pass_kwargs = (kwargs && kwargs->d.size());
+    Box* key_func;
+    Box* extremElement;
     Box* container;
+    Box* extremVal;
+
+    if (pass_kwargs) {
+        static BoxedString* key_str = static_cast<BoxedString*>(PyString_InternFromString("key"));
+        auto it = kwargs->d.find(key_str);
+        key_func = it->second;
+    }
+
     if (args->size() == 0) {
-        minElement = nullptr;
+        extremElement = nullptr;
+        extremVal = nullptr;
         container = arg0;
     } else {
-        minElement = arg0;
+        extremElement = arg0;
+        if (pass_kwargs) {
```

Comment:
As a minor stylistic thing, I think it'd be nicer if this was `if (key_func)` -- it just so happens that the conditions will be true at the same time, but checking `key_func` seems like better one to be checking.